tests/core/arch-arm64/arm64stalker-fixture.c

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright (C) 2009-2013 Ole André Vadla Ravnås <ole.andre.ravnas@tillitech.com>
+ * Copyright (C) 2010-2013 Karl Trygve Kalleberg <karltk@boblycat.org>
+ *
+ * Licence: wxWindows Library Licence, Version 3.1
+ */
+
+#include "gumstalker.h"
+
+#include "fakeeventsink.h"
+#include "gumarm64writer.h"
+#include "gummemory.h"
+#include "testutil.h"
+
+#include <stdlib.h>
+#include <string.h>
+
+#define STALKER_ARM64_TESTCASE(NAME) \
+    void test_arm64_stalker_ ## NAME ( \
+    TestArm64StalkerFixture * fixture, gconstpointer data)
+#define STALKER_ARM64_TESTENTRY(NAME) \
+    TEST_ENTRY_WITH_FIXTURE ("Core/Arm64Stalker", test_arm64_stalker, NAME, \
+    TestArm64StalkerFixture)
+
+#define NTH_EVENT_AS_CALL(N) \
+    (gum_fake_event_sink_get_nth_event_as_call (fixture->sink, N))
+#define NTH_EVENT_AS_RET(N) \
+    (gum_fake_event_sink_get_nth_event_as_ret (fixture->sink, N))
+#define NTH_EXEC_EVENT_LOCATION(N) \
+    (gum_fake_event_sink_get_nth_event_as_exec (fixture->sink, N)->location)
+
+typedef struct _TestArm64StalkerFixture
+{
+  GumStalker * stalker;
+  GumFakeEventSink * sink;
+
+  guint8 * code;
+  guint8 * last_invoke_calladdr;
+  guint8 * last_invoke_retaddr;
+} TestArm64StalkerFixture;
+
+typedef gint (* StalkerTestFunc) (gint arg);
+
+static void silence_warnings (void);
+
+static void
+debug_hello (gpointer pointer)
+{
+  g_print ("* pointer: %p *\n", pointer);
+}
+
+static void
+put_debug_print (GumArm64Writer * cw,
```

Comment:
Suggest `put_debug_print` => `put_debug_print_pointer` for consistency with `_reg()`.